### PR TITLE
feat(content): add support for accessing content files within nested folders

### PR DESCRIPTION
**Pull request by BaronVonPerko**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-angular-plugin
- [ ] astro-angular
- [ ] create-analog
- [ ] router
- [ ] platform
- [x] content

## What is the current behavior?

The current `injectContent()` does not support subdirectories within the **content** folder.

Issue Number: #282 

## What is the new behavior?

This allows the existing string parameter for the routeParam to remain the same, but also allows the developer to pass in an object instead, consisting of the routeParam key as well as the subdirectory name.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/docs-app/docs/features/routing/content.md [hunk 1]
```diff
@@ -191,3 +191,49 @@ export default class BlogPostComponent {
...
+
+## Support for Content Subdirectories
+
+Analog also supports subdirectories within your content folder.
+
+The `injectContent()` function can also be used with an object that contains the route parameter and subdirectory name.
+
+This can be useful if, for instance, you have blog posts, as well as a portfolio of project markdown files to be used on the site.
+
+```ts
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/routing/content.md:203]: Using a treeview would fit better here. Example below

```treeview
src/
└── app/
    └── routes/
        │   └── products/
        │      ├──[productId].ts
        │      └──(products-list).ts
        └── products.ts
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: apps/docs-app/docs/features/routing/content.md (hunk 1 above)
Comment (BaronVonPerko) [reply apps/docs-app/docs/features/routing/content.md:203]: Updated 👍🏻

**Review by brandonroberts — APPROVED** [review]
LGTM

**Comment by brandonroberts** [comment]
Awesome! @allcontributors add @BaronVonPerko for code,doc

**Pull request merged**